pkgs/applications/editors/pulsar/update.mjs

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env nix-shell
+/*
+#!nix-shell -i node -p nodejs-18_x
+*/
+
+import { promises as fs } from 'node:fs';
+import { promisify } from 'node:util';
+import { exec as _exec } from 'node:child_process';
+const exec = promisify(_exec);
+
+const constants = {
+    githubUrl: "https://api.github.com/repos/pulsar-edit/pulsar/releases",
+    sha256FileURL: (newVersion) => `https://github.com/pulsar-edit/pulsar/releases/download/v${newVersion}/SHA256SUMS.txt`,
+    x86_64FileName: (newVersion) => `Linux.pulsar-${newVersion}.tar.gz`,
+    aarch64FileName: (newVersion) => `ARM.Linux.pulsar-${newVersion}-arm64.tar.gz`,
+};
+
+async function getLatestVersion() {
+    const requestResult = await fetch(constants.githubUrl);
+    if (!requestResult.ok) {
+        console.error("Failed to fetch releases");
+        console.error(requestResult);
+        process.exit(1);
+    };
+    let jsonResult = await requestResult.json();
+
+    jsonResult = jsonResult.filter((release) => !release.prerelease && !release.draft);
+    if (jsonResult.length == 0) {
+        console.error("No releases found");
+        process.exit(1);
+    }
+
+    return jsonResult[0].tag_name.replace(/^v/, '');
+}
+
+async function getSha256Sum(hashFileContent, targetFile) {
+    // The upstream file has a fomat like this:
+    // 0000000000000000000000000000000000000000000000000000000000000000 targetFile
+
+    let targetHash = hashFileContent.
+        split('\n').
+        filter((line) => line.endsWith(targetFile))[0].
+        split(' ')[0];
+
+    return exec(`nix --experimental-features nix-command hash to-sri --type sha256 "${targetHash}"`). // Convert the hash to SRI format
```

Comment:
sri is just sha256- + Buffer.from(hash, 'hex').toString('base64'), so we could save the nix call here

Reply:
Oh yes. Somehow I couldn't make this work beforehand when basing myself on `nix-prefetch-url`. It's fixed now, thanks :)